diff --git a/src/EasyBreadcrumbBuilder.php b/src/EasyBreadcrumbBuilder.php
index baa93c1e..36a41f7a 100644
--- a/src/EasyBreadcrumbBuilder.php
+++ b/src/EasyBreadcrumbBuilder.php
@@ -280,6 +280,13 @@ public function build(RouteMatchInterface $route_match) {
     // resolving path aliases so the breadcrumb can be defined by creating a
     // hierarchy of path aliases.
     $path = trim($this->context->getPathInfo(), '/');
+
+    // Ensure that Views AJAX requests do not seep into the breadcrumb.  This
+    // can be a problem when the breadcrumb exists inside the view header.
+    if($path == 'views/ajax') {
+      $currentpath = \Drupal::service('path.current')->getPath();
+      $path = trim(\Drupal::service('path_alias.manager')->getAliasByPath($currentpath), '/');
+    }
     $path = urldecode($path);
     $path_elements = explode('/', $path);
     $front = $this->siteConfig->get('page.front');
