diff --git a/permissions_by_term.module b/permissions_by_term.module
index 6597a7a..ed1e538 100644
--- a/permissions_by_term.module
+++ b/permissions_by_term.module
@@ -205,6 +205,7 @@ function permissions_by_term_form_taxonomy_term_form_alter(&$form, FormStateInterface $formState, $form_id) {
 
     // Generate the default values for the form.
     $aSetRoles = [];
+    array_push($aSetRoles,"administrator");
     if (!empty($aTranslatedAllowedRoleNames)) {
       foreach ($aTranslatedAllowedRoleNames as $role_name) {
         $aSetRoles[] = $role_name;
